tiledb/sm/query/query_condition.h

```diff
@@ -84,27 +83,33 @@ class QueryCondition {
     Clause(const Clause& rhs)
         : field_name_(rhs.field_name_)
         , condition_value_data_(rhs.condition_value_data_)
-        , condition_value_(
-              rhs.condition_value_ == nullptr ? nullptr :
-                                                condition_value_data_.data())
+        , condition_value_view_(
+              (rhs.condition_value_view_.content() == nullptr ?
+                   nullptr :
+                   condition_value_data_.data()),
+              condition_value_data_.size())
         , op_(rhs.op_){};
 
     /** Move constructor. */
     Clause(Clause&& rhs)
         : field_name_(std::move(rhs.field_name_))
         , condition_value_data_(std::move(rhs.condition_value_data_))
-        , condition_value_(
-              rhs.condition_value_ == nullptr ? nullptr :
-                                                condition_value_data_.data())
+        , condition_value_view_(
```

Comment:
Since this is a move constructor, shouldn't we just move the UntypedDatum view object into the new Clause object? Why create a new object?